eos-updater-prepare-volume/eos-updater-prepare-volume

```diff
@@ -0,0 +1,228 @@
+        flatpak_collection_refs = list(zip(refs_iter, refs_iter))
+        invalid_refs = [collection_ref
+                        for collection_ref in flatpak_collection_refs
+                        if not self._validate_flatpak_ref(collection_ref)]
+        if invalid_refs:
+            refs_list = (', '.join(['(%s, %s)' %
+                                    (collection_id, ref)
+                                    for (collection_id, ref) in invalid_refs]))
+            return self.__fail(self.EXIT_INVALID_ARGUMENTS,
+                               'Invalid flatpak collection–refs: %s' % refs_list)
+
+        # FIXME: Do we also want to pull in related refs, like locales?
+        # Currently, they can be listed explicitly on the command line.
+        runtime_collection_refs = \
+            self._get_runtimes_for_flatpaks(flatpak_collection_refs)
+
+        os_collection_ref = self._get_os_collection_ref()
+        if not os_collection_ref:
+            return self.__fail(self.EXIT_FAILED,
+                               'OSTree deployment ref could not be found; '
+                               'are you on an OSTree system?')
+
+        print(list(flatpak_collection_refs))
+        collection_refs = \
+            [os_collection_ref] + \
+            list(flatpak_collection_refs) + \
+            list(runtime_collection_refs)
+
+        # Eliminate duplicates.
+        collection_refs = list(set(collection_refs))
+
+        # Pass the heavy lifting off to `ostree create-usb`.
+        # TODO: Verify that it adds GPG keys where appropriate.
+        # TODO: Verify that a summary file is present.
+        flattened_collection_refs = [e for l in collection_refs for e in l]
+        (_, repo) = self.sysroot.get_repo()
+        self.__run(['ostree', 'create-usb',
+                    '--repo', os.path.realpath(repo.get_path().get_path()),
+                    self.volume_path] +
+                   flattened_collection_refs)
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Prepare a USB drive with a copy of the local OSTree '
+                    'repository and the specified flatpaks, so it can be '
+                    'used to update other machines offline. The repository '
+                    'copy will be put in the .ostree/repo directory on the '
+                    'USB drive; other files will not be affected.')
+    parser.add_argument('volume_path', metavar='VOLUME-PATH', type=str,
+                        help='path to the USB drive to prepare')
+    parser.add_argument('--quiet', action='store_const', const=True,
+                        help='do not print anything; check exit status '
+                             'for success')
+    parser.add_argument('flatpak_refs', metavar='FLATPAK-REF', nargs='*',
+                        help='refs of flatpaks to put on the USB drive')
+
+    args = parser.parse_args()
+
+    VolumePreparer(**vars(args)).prepare_volume()
```

Comment:
I was surprised that mapping kwargs to named positionals works, though this seems somewhat fragile to me (since there's now a tight coupling between the argument names and the names of the parameters to the positionals).

Reply:
Yup, I am happy with that tight coupling, given that the class is basically just designed for use by this command line tool. The coupling can be changed easily enough if the class is factored out to be used elsewhere.